### PR TITLE
Should fix the exception with commands outside of the dev environment.

### PR DESCRIPTION
**Pull request by Raycoms**
Fixes the exception.

Review please @Minecolonies/maintainers

### CODE REVIEW

**Comment by Raycoms** [comment]
Have to test this first.

File: src/main/java/com/minecolonies/commands/MinecoloniesCommand.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.minecolonies.commands;
 
 import com.google.common.collect.ImmutableMap;
+import net.minecraft.command.CommandBase;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/MinecoloniesCommand.java]: MINOR Remove this unused import 'net.minecraft.command.CommandBase'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* MINOR 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractSplitCommand.java#L28](https://github.com/Minecolonies/minecolonies/blob/90298da7bfe12ce5534abf9aed5a58208ef61ef6/src/main/java/com/minecolonies/commands/AbstractSplitCommand.java#L28): It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* MINOR 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractSplitCommand.java#L28](https://github.com/Minecolonies/minecolonies/blob/a09995f9f26d225cadac78ca50e6e8f85ede54d5/src/main/java/com/minecolonies/commands/AbstractSplitCommand.java#L28): It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 4 issues
* MINOR 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractSplitCommand.java#L28](https://github.com/Minecolonies/minecolonies/blob/a5110a710db606c1608498e4450b5ecd7edae6f3/src/main/java/com/minecolonies/commands/AbstractSplitCommand.java#L28): It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)
1. MINOR [ColoniesCommand.java#L17](https://github.com/Minecolonies/minecolonies/blob/a5110a710db606c1608498e4450b5ecd7edae6f3/src/main/java/com/minecolonies/commands/ColoniesCommand.java#L17): Define a constant instead of duplicating this literal "colonies" 3 times. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* MINOR 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [ColoniesCommand.java#L20](https://github.com/Minecolonies/minecolonies/blob/c59449772fc06d1fbf7371830f0504dd787b3477/src/main/java/com/minecolonies/commands/ColoniesCommand.java#L20): Define a constant instead of duplicating this literal "colonies" 3 times. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java [hunk 2]
```diff
@@ -11,23 +16,44 @@
  */
 public class ColoniesCommand extends AbstractSplitCommand
 {
+    public static final String desc = "colonies";
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00115)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* MINOR 3 minor

Watch the comments in this conversation to review them.

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 4 issues
* MINOR 4 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractSingleCommand.java#L20](https://github.com/Minecolonies/minecolonies/blob/6a6a01969b5da777d37cfcb883dce7f922c746ed/src/main/java/com/minecolonies/commands/AbstractSingleCommand.java#L20): It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 4 issues
* MINOR 4 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractSingleCommand.java#L20](https://github.com/Minecolonies/minecolonies/blob/8a930709f2ea3d614e4dd78d6f1b4fb04a183606/src/main/java/com/minecolonies/commands/AbstractSingleCommand.java#L20): It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 4 issues
* MINOR 4 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractSingleCommand.java#L20](https://github.com/Minecolonies/minecolonies/blob/e4e34b5511a9b36e9cb0b32bfdf85c6529d60c86/src/main/java/com/minecolonies/commands/AbstractSingleCommand.java#L20): It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)

File: src/main/java/com/minecolonies/commands/AbstractSplitCommand.java [hunk 3]
```diff
@@ -14,7 +15,7 @@
 /**
  * A command that has children. Split-command with various parts.
  */
-public abstract class AbstractSplitCommand implements ISubCommand
+public abstract class AbstractSplitCommand extends AbstractSingleCommand implements ISubCommand
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/AbstractSplitCommand.java]: MINOR "ISubCommand" is implemented by a super class; there is no need to implement it here. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1939)

File: src/main/java/com/minecolonies/commands/AbstractSplitCommand.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package com.minecolonies.commands;
 
+import net.minecraft.command.CommandBase;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/AbstractSplitCommand.java]: MINOR Remove this unused import 'net.minecraft.command.CommandBase'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java [hunk 5]
```diff
@@ -11,23 +17,50 @@
...
       new ImmutableMap.Builder<String, ISubCommand>()
-        .put("list", new ListColonies("mc", "colonies", "list"))
+        .put("list", new ListColonies("minecolonies", DESC, "list"))
         .build();
 
     /**
      * Initialize this command with it's parents.
      */
-    public ColoniesCommand()
+    public ColoniesCommand(@NotNull final String parent)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Document the parameter(s): parent [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 7 issues
* MINOR 7 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractSingleCommand.java#L20](https://github.com/Minecolonies/minecolonies/blob/86ca98cc6149b31343b997c7517a938c3f6acab5/src/main/java/com/minecolonies/commands/AbstractSingleCommand.java#L20): It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)

File: src/main/java/com/minecolonies/commands/MinecoloniesCommand.java [hunk 6]
```diff
@@ -1,6 +1,7 @@
...
 import net.minecraft.command.ICommand;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.server.MinecraftServer;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/MinecoloniesCommand.java]: MINOR Remove this unused import 'net.minecraft.server.MinecraftServer'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/CommandEntryPoint.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.List;
+
+/**
+ * Command entry point to make minecraft inheritance happy.
+ */
+public class CommandEntryPoint extends CommandBase
+{
+
+    @NotNull
+    private MinecoloniesCommand root;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/CommandEntryPoint.java]: MAJOR Private field 'root' could be made final; it is only initialized in the declaration or constructor. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AImmutableField)

File: src/main/java/com/minecolonies/commands/CommandEntryPoint.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+    /**
+     * Create our entry point once.
+     */
+    public CommandEntryPoint()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/CommandEntryPoint.java]: MINOR It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java [hunk 9]
```diff
@@ -1,7 +1,13 @@
 package com.minecolonies.commands;
 
 import com.google.common.collect.ImmutableMap;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Remove this unused import 'net.minecraft.command.ICommandSender'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java [hunk 10]
```diff
@@ -1,7 +1,13 @@
...
+import net.minecraft.server.MinecraftServer;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Remove this unused import 'net.minecraft.server.MinecraftServer'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java [hunk 11]
```diff
@@ -1,7 +1,13 @@
...
+import org.jetbrains.annotations.NotNull;
 
+import java.util.Arrays;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Remove this unused import 'java.util.Arrays'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java [hunk 12]
```diff
@@ -1,7 +1,13 @@
...
+import java.util.List;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Remove this unused import 'java.util.List'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 15 issues
* MAJOR 1 major
* MINOR 14 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [MinecoloniesCommand.java#L8](https://github.com/Minecolonies/minecolonies/blob/32841df6521962d66c0e0271f70c69e2224e47f9/src/main/java/com/minecolonies/commands/MinecoloniesCommand.java#L8): Remove this unused import 'org.jetbrains.annotations.NotNull'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [MinecoloniesCommand.java#L10](https://github.com/Minecolonies/minecolonies/blob/32841df6521962d66c0e0271f70c69e2224e47f9/src/main/java/com/minecolonies/commands/MinecoloniesCommand.java#L10): Remove this unused import 'java.util.Arrays'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [MinecoloniesCommand.java#L11](https://github.com/Minecolonies/minecolonies/blob/32841df6521962d66c0e0271f70c69e2224e47f9/src/main/java/com/minecolonies/commands/MinecoloniesCommand.java#L11): Remove this unused import 'java.util.List'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/MinecoloniesCommand.java (hunk 6 above)
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/MinecoloniesCommand.java]: MINOR Remove this unused import 'net.minecraft.server.MinecraftServer'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/CommandEntryPoint.java (hunk 8 above)
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/CommandEntryPoint.java]: MINOR It is a good practice to call super() in a constructor [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java (hunk 9 above)
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Remove this unused import 'net.minecraft.command.ICommandSender'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java (hunk 10 above)
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Remove this unused import 'net.minecraft.server.MinecraftServer'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java (hunk 11 above)
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Remove this unused import 'java.util.Arrays'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/commands/ColoniesCommand.java (hunk 12 above)
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/commands/ColoniesCommand.java]: MINOR Remove this unused import 'java.util.List'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by Kostronor** [comment]
:shipit: @cltnschlosser @Raycoms

**Comment by Raycoms** [comment]
:shipit:

**Comment by cltnschlosser** [comment]
:shipit:

**Pull request merged**